frontend/src/main/scala/bloop/testing/TestSuiteEvent.scala

```diff
@@ -63,38 +59,40 @@ class LoggingEventHandler(logger: Logger) extends TestSuiteEventHandler {
       val pending = events.count(_.status() == Status.Pending)
       val errors = events.count(_.status() == Status.Error)
 
-      events
+      // Log any exception that may have happened
+      events.iterator
         .filter(_.throwable().isDefined)
         .map(_.throwable().get())
         .foreach(logger.trace)
 
       logger.info(s"Execution took ${TimeFormat.humanReadable(duration)}.")
-
-      val regularMetrics = List(testsTotal -> "tests",
-                                passed -> "passed",
-                                pending -> "pending",
-                                ignored -> "ignored",
-                                skipped -> "skipped")
-
+      val regularMetrics = List(
+        testsTotal -> "tests",
+        passed -> "passed",
+        pending -> "pending",
+        ignored -> "ignored",
+        skipped -> "skipped"
+      )
+
+      val failureCount = failed + canceled + errors
       val failureMetrics = List(failed -> "failed", canceled -> "canceled", errors -> "errors")
-
       logger.info(formatMetrics(regularMetrics ++ failureMetrics))
 
-      if (failureMetrics.exists(_._1 != 0)) suitesFailed.append(testSuite)
+      if (failureCount > 0) suitesFailed.append(testSuite)
       else {
         suitesPassed += 1
-        logger.info(s"All tests passed.")
+        logger.info(s"All tests in ${testSuite} passed.")
       }
 
       logger.info("")
-
       suitesTotal += 1
       suitesDuration += duration
 
-    case TestSuiteEvent.Done =>
+    case TestSuiteEvent.Done => ()
   }
 
-  def report(): Unit = {
+  override def report(): Unit = {
+    // TODO: Shall we think of a better way to format this delimiter based on screen length?
```

Comment:
Good idea, however determining the terminal width may not always be possible. We could count the number of characters of the longest row instead.